web/src/todo_widget.ts

```diff
@@ -213,6 +213,31 @@ export class TaskData {
                 item.completed = !item.completed;
             },
         },
+
+        reorder: {
+            inbound: (_sender_id: number, raw_data: unknown): void => {
+                const task_reorder_inbound_data_schema = z.object({
+                    type: z.literal("reorder"),
+                    from: z.number(),
+                    to: z.number(),
+                });
+                const parsed = task_reorder_inbound_data_schema.safeParse(raw_data);
+                if (!parsed.success) {
+                    blueslip.warn("todo widget: bad type for inbound reorder data", {
+                        error: parsed.error,
+                    });
+                    return;
+                }
+
+                const tasks = [...this.task_map.entries()];
+                const [movedTask] = tasks.splice(parsed.data.from, 1);
+                if (!movedTask) {
+                    return;
```

Comment:
Is this expected to never happen? Or could a buggy client send invalid `to` / `from` values? I feel like we should be preventing this possibility in earlier validation.